### DISCUSSION TITLE
Flask 10.29 has been released! Look here for breaking changes.

### DISCUSSION THREAD

**Discussion by FrederikBolding** [MetaMask/snaps#1364] [Announcements]
Flask `10.29.0` has been released to Chrome (Firefox pending). See below for breaking changes:

- Snaps can no longer request `eth_requestAccounts` or `wallet_requestSnaps`
- Snaps are now required to request permission for at least one handler permission, e.g. `onRpcRequest`

Keep in mind that since `10.28.2` your CLI (`@metamask/snaps-cli`) must be at least version `0.30.0`.

**Comment by zwilling** [comment]
Thanks a lot for the update.

I have a question about removing access to `eth_requestAccounts`:
Is it intendet that Snaps can still use the following?
```
await ethereum.request({
  method: 'wallet_requestPermissions',
  params: [{ eth_accounts: {} }],
});
```
Or is this possibility also going to be removed soon?

Asking because it achieves the same for us.
We use a signature by the user's (HW) wallet to derive an EdDSA key for use in zero knowledge proofs.
Requesting a signature once seems less invasive than using the Bip32Entropy permission. Especially if the user has a HW wallet.

**Reply by Montoya** [reply]
Hi @zwilling, I apologize, we will have to remove access to that too. Snaps permissions are not meant to be dynamic and this essentially allows a snap to request dynamic permissions.

**Reply by neekolas** [reply]
So, what permission do I need to enable to get a wallet signature (and to get the current wallet address) from inside a Snap. That is core to being able to have a snap sign in to XMTP.

**Reply by zwilling** [reply]
Hi @Montoya ,

Thanks for the information. I would like to ask for a clarification on this.

1. According to https://github.com/MetaMask/snaps/discussions/1411 the API is not planned to be changed although `wallet_requestPermissions` still works. Are you still planning to remove access to `wallet_requestPermissions`?
2. Similar to @neekolas , getting a signature from inside a Snap would be important for us to derive a key. I would like to use https://docs.metamask.io/snaps/reference/rpc-api/#snap_getentropy instead, but it does not seem to work with hardware wallets or imported accounts. Is there an alternative way we could use that does not require passing the signature through the web site front-end?

**Reply by Montoya** [reply]
@zwilling 
1. Yes we are, we have a version of the extension waiting for release that has it removed. 
2. Unfortunately no. We are working on a new API that will solve this but it will come after the launch of Snaps to the extension.

**Reply by zwilling** [reply]
Thanks a lot. I appreciate the early warning and will work on an alternative solution.
